### ISSUE TITLE
TypeORM unexpectedly loading migrations

### ISSUE THREAD

**Issue by r-walsh** [typeorm/typeorm#3277]
**Issue type:**

[x] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[x] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.9` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I'm having an issue where TypeORM is loading my migration files on server start. Because I keep my migrations as `.ts` files and run them using `ts-node` with the ORM CLI this causes the server to crash. I've set `migrationsRun: false` in `ormconfig.js` which I expected to avoid the problem. Is there a way to prevent TypeORM from loading the migration files on server start?

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by havenchyk** [comment]
@r-walsh could you please create a small repo to show the problem? There is a bit of general advice not to compile your `.js` files to the same folder with `.ts`, can it be the problem?

**Comment by pleerock** [comment]
it shouldn't run migrations if migrationsRun is set to false for sure, please provide a reproduction

**Comment by bkoninski-neducatio** [comment]
I have the same problem. Application is written in JavaScript. Migrations are being run in CLI using ts-node - this part works great. But server starting fails trying to read *.ts migrations files.

**Comment by pleerock** [comment]
@r-walsh I checked your repository. You messed it. You are importing `.ts` files in your ormconfig while you are running js file (e.g. `node dist/index.js`). JAVASCRIPT TRIES TO IMPORT TYPESCRIPT FILES - THATS WHY YOU HAVE ERROR.

Use `.ts` ONLY IF YOU ARE USING TS-NODE. If you aren't using ts-node you should only use `.js` files everywhere since without ts-node javascript have no idea what are your `.ts` files are.

**Comment by r-walsh** [comment]
I understand that node cannot run `.ts` files, this is why in my initial issue I ask

> Is there a way to prevent TypeORM from loading the migration files on server start?

Everything about the workflow in the repo works except for TypeORM pulling in the migration files causing a crash due to them being in TS. I am only asking if there is a way to prevent TypeORM from loading those migration files on startup (since I don't want to run migrations on startup anyway)

**Comment by geritol** [comment]
Just ran into the exact same issue.
Typeorm should not try to import/run files under `migrations` when `migrationsRun: false` is specified

**Comment by LeoFC97** [comment]
Im still having this problem, migrationsRun: false dosnt work

**Comment by thomazmz** [comment]
I'm currently having the above problem in a project that am working on. 

Yes, my `.ormconfig` file does reference `.ts` files on `migrations` glob and this works perfectly when running the project with `ts-node` as suggested. This behavior is definitely fine and approved.

However what I do not understand is why `typeorm` internally tries to import those `.ts` files even when the `migrationsRun` options is set false. 

I totally know that it will not work if you try to run your application with `migrationsRun` option turned on and you have `.ts` files referenced on the glob, this is Node JS in nature behaving as expected. 

What I do not knowI is why TypeORM could not simply ignore importing those migration files if `migrationsRun` option is set false. This would prevent the problem we are facing from happening. 

In other words: I don't see the point why the library tries to import the migration files if they are not intended to run anyway. 

@pleerock If there is any technical information I might be missing on this context that prevents the suggested behavior from being implemented I would be very happy to know.

Thanks in advance.

**Comment by RuslanUshakov** [comment]
Any updates on this issue? I face with the same one
